### DISCUSSION TITLE
Crop images support

### DISCUSSION THREAD

**Discussion by htmgarcia** [publishpress/publishpress-blocks#1203] [Feature Requests]
Suggested through https://publishpress.com/blog/publishpress-blocks/content-display/#comment-51603

> [ ...] automatically limit images to a certain size or aspect ratio, and auto-crop ones that don’t fit [...]

**Comment by contactashish13** [comment]
@htmgarcia Just wanted to find out if I understood the requirements correctly. I have 2 posts that have images of different sizes (tall image vs. regular sized) and in the content block they show misaligned like this:

![image](https://user-images.githubusercontent.com/12953439/124711775-1d626c80-df1c-11eb-9574-ad58e8fc5d4f.png)

The intent is to make them align properly by cropping the image so that they align like this:

![image](https://user-images.githubusercontent.com/12953439/124712766-5bac5b80-df1d-11eb-9d23-f7eb3c2054a8.png)

Is that correct?

Also, how do you see this working in the block editor? Maybe a text control in the inspector that defines the height of the image box? Let me know if have seen any example blocks whose UX you like.

**Comment by htmgarcia** [comment]
@contactashish13 to start, I see a couple of scenarios in this request:
- Option to crop images to have the exact same width and height
- Option to resize images keeping aspect ratio by changing width or height

About the UX, I would think in a simple way to resize/crop images by using a panel in the sidebar with  few controls.
![Screen Shot 2021-07-08 at 14 07 28](https://user-images.githubusercontent.com/4999794/124977234-dcd72180-dff5-11eb-8b19-8946a30ca54d.png)
![Screen Shot 2021-07-08 at 14 07 18](https://user-images.githubusercontent.com/4999794/124977237-dd6fb800-dff5-11eb-9fa1-055df72a3247.png)

This approach can have a problem with some views such as Frontpage and Newspaper due some items display wider content than others.

**Comment by htmgarcia** [comment]
@contactashish13 any luck with this one?

**Comment by contactashish13** [comment]
@htmgarcia Sorry, I intend to work on this this week time permitting :)

**Comment by htmgarcia** [comment]
@contactashish13 let's pause this issue, please.

**Comment by htmgarcia** [comment]
Let's check in deep the Image block crop feature to see if we can integrate with Content Display.
